docs/content/latest/api/ysql/exprs/func_yb_hash_code.md

```diff
@@ -0,0 +1,195 @@
+
+### Finding the hash code of a particular set of values
+Given a set of expressions, we can compute the hash code of them directly as such
+
+```
+SELECT yb_hash_code(1::int, 2::int, ‘sample string’::text);
+
+   yb_hash_code
+---------------------
+      5067
+```
+
+### Querying rows based on physically hash sharded location
+We can request a batch of rows from a tablet of our choice without unnecessarily touching other tablets.
+
+```
+SELECT * FROM sample_table WHERE yb_hash_code(x,y) >= 0 and yb_hash_code(x,y) <= 128;
+```
+
+Assuming each tablet holds at least 128 hash values, this statement will request rows from the first tablet of `sample_table`. Similarly, we can request a batch of physically colocated rows based on index table location too.
+
+```
+SELECT * FROM sample_table WHERE yb_hash_code(x,z) >= 0 and yb_hash_code(x,z) <= 128;
+```
+
+### Querying a random batch of rows
+We can extend the above use case to sample a random batch of rows by selecting over a random fixed size interval over the hash partition space. For example, we select over a random space of 512 hash values. Note that this is 1/128 of the total hash partition space as there are 2^16 = 65536 hash values in total. We first select a random lower bound for this interval.
+
+```
+SELECT floor(random()*65536);
+
+ floor
+-------
+ 4600
+```
+
+Now, we can search over all rows whose hash values range in [4600, 5112). In this example, we take a count of all such rows.
+
+```
+SELECT COUNT(*) FROM sample_table WHERE yb_hash_code(x,y) >= 4600 and yb_hash_code(x,y) < 5112;
+
+ count 
+-------
+    74
+(1 row)
+```
+
+Since, we use what we assume to be a uniformly distributed hash function to partition our rows, we can assume that this is a count of approximately 1/128 of all the rows. Therefore, multiplying this count, 78 by 128 gives us a good estimate of the total number of rows (9984 in this case) in the table without querying and iterating over all tablets.
+
+### Distributed Parallel Queries
+Seeing how we can constrain our queries to a physically collocated group of rows of the user’s choosing now we can shard individual aggregate queries across multiple partition groups. These sharded queries can execute in parallel. For example, we can execute a batch of `COUNT(*)` queries as follows on separate threads:
+
+```
+SELECT COUNT(*) FROM sample_table WHERE yb_hash_code(x,y) >= 0 and yb_hash_code(x,y) < 8192;
+
+
+SELECT COUNT(*) FROM sample_table WHERE yb_hash_code(x,y) >= 8192 and yb_hash_code(x,y) < 16384;
+
+SELECT COUNT(*) FROM sample_table WHERE yb_hash_code(x,y) >= 16384 and yb_hash_code(x,y) < 24576;
+.
+.
+.
+.
+
+SELECT COUNT(*) FROM sample_table WHERE yb_hash_code(x,y) >= 57344 and yb_hash_code(x,y) < 65536;
+```
+
+Summing up all the results of these queries gives us the equivalent of a `COUNT(*)` over all of `sample_table`.
```

Comment:
nit: once your script lands to yb-tools maybe we can reference that somewhere in this section?